### PR TITLE
config: Improve config migrate messaging.

### LINKED ISSUE #4199: Obscure message when starting minio server
Error message printed was not comprehensible. 

## Expected Behavior
It would just start and print the usual access key/secret key info

## Current Behavior
Migration from 11 to 18 was a series of messages seen on startup. These numbers looked different from minio server's version information leading me to wonder what was actually being migrated. If it was anything internal, does the user even need to know about it?

## Possible Solution
Evaluate if there is a valid need to show config file migration messages to the user.  

## Steps to Reproduce (for bugs)
Install an older version of minio first. Then install a much newer version of the binary. You should be able to see these messages. 

## Context
 It's a cosmetic issue. 

## Your Environment

* Version used (`minio version`):  2017-04-28T19:24:01Z
* Environment name and version (e.g. nginx 1.9.1): none
* Server type and version: fs
* Operating System and version (`uname -a`): 
* Link to your project: 16.3.0 Darwin Kernel Version 16.3.0

Actual message :
```
minio server ~/Downloads/miniodata/
Migration from version ‘10’ to ‘11’ completed successfully.
Migration from version ‘11’ to ‘12’ completed successfully.
Migration from version ‘12’ to ‘13’ completed successfully.
Migration from version ‘13’ to ‘14’ completed successfully.
Migration from version ‘14’ to ‘15’ completed successfully.
Migration from version ‘15’ to ‘16’ completed successfully.
Migration from version ‘16’ to ‘17’ completed successfully.
Migration from version ‘17’ to ‘18’ completed successfully.

Endpoint:  http://192.168.1.9:9000  http://127.0.0.1:9000
AccessKey:  <redacted>
SecretKey:   <redacted>
Region:    us-east-1
SQS ARNs:  <none>

Browser Access:
   http://192.168.1.9:9000  http://127.0.0.1:9000

```

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Previous message

```
Migration from version ‘17’ to ‘18’ completed successfully.
```

For example didn't provide any meaningful insights.

This PR attempts to improve this message as below

```
Configuration file '/home/harsha/.minio/config.json' migrated from version '17' to '18' successfully.
```

## Motivation and Context
Fixes #4199

## How Has This Been Tested?
Manually observing the message.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cmd/config-migrate.go [hunk 1]
```diff
@@ -24,6 +24,9 @@ import (
 	"github.com/minio/minio/pkg/quick"
 )
 
+// DO NOT EDIT following message template, please open a github issue to discuss instead.
+var configMigrateMSGTemplate = "Minio configuration file %s migrated from version '%s' to '%s' successfully.\n"
```
Comment (balamurugana) [inline cmd/config-migrate.go]: I think `Configuration file` is good enough than `Minio configuration file`.  As minio server shows this at startup `Minio` is redundant word.

Comment (harshavardhana) [reply cmd/config-migrate.go]: Good idea. done.

**Review by balamurugana — APPROVED** [review]
Verified

**Review by donatello — APPROVED** [review]
:+1:

**Pull request merged**